### PR TITLE
add support for multiple private and public keys

### PR DESCRIPTION
**Pull request by luitzifa**
I needed a way to define multiple keys for one user.

### CODE REVIEW

**Comment by aboe76** [comment]
could you please add pillar.example for this.

File: users/init.sls [hunk 1]
```diff
@@ -170,35 +170,44 @@ user_keydir_{{ name }}:
...
+      {% else %}
+        {% set key_name = _key %}
+      {% endif %}
+users_{{ name }}_{{ key_name }}_private_key:
   file.managed:
-    - name: {{ home }}/.ssh/{{ key_type }}
+    - name: {{ home }}/.ssh/{{ key_name }}
     - user: {{ name }}
     - group: {{ user_group }}
     - mode: 600
```
Comment (aboe76) [inline users/init.sls]: this will be used twice

File: users/init.sls [hunk 2]
```diff
@@ -170,35 +170,44 @@ user_keydir_{{ name }}:
...
       - group: users_{{ name }}_{{ group }}_group
       {% endfor %}
-users_user_{{ name }}_public_key:
+users_{{ name }}_{{ key_name }}_public_key:
   file.managed:
-    - name: {{ home }}/.ssh/{{ key_type }}.pub
+    - name: {{ home }}/.ssh/{{ key_name }}
     - user: {{ name }}
     - group: {{ user_group }}
     - mode: 644
```
Comment (aboe76) [inline users/init.sls]: this wil be used twice..

**Comment by luitzifa** [comment]
yeah, that was stupid

**Comment by aboe76** [comment]
@luitzifa, yes thanks for updating the PR, only issue left is are we ok with changing the pubkey permissions from 644 to 600? I don't think this will matter a lot but maybe someone has a use case with reading pubkeys from other users?

**Comment by luitzifa** [comment]
this should work

**Comment by aboe76** [comment]
thanks @luitzifa

**Pull request merged**